src/commands/publish/mod.rs

```diff
@@ -65,16 +39,48 @@ fn multi_script(user: &User, name: &str) -> Result<(), failure::Error> {
     let client = reqwest::Client::new();
     let settings = user.settings.clone();
 
-    client
-        .put(&worker_addr)
-        .header("X-Auth-Key", settings.global_user.api_key)
-        .header("X-Auth-Email", settings.global_user.email)
-        .multipart(build_form()?)
-        .send()?;
+    let mut res = match project_type {
+        ProjectType::Rust => {
+            info!("Rust project detected. Publishing...");
+            client
+                .put(&worker_addr)
+                .header("X-Auth-Key", settings.global_user.api_key)
+                .header("X-Auth-Email", settings.global_user.email)
+                .multipart(build_form()?)
+                .send()?
+        }
+        ProjectType::JavaScript => {
+            info!("JavaScript project detected. Publishing...");
+            client
+                .put(&worker_addr)
+                .header("X-Auth-Key", settings.global_user.api_key)
+                .header("X-Auth-Email", settings.global_user.email)
+                .header("Content-Type", "application/javascript")
+                .body(build_js()?)
+                .send()?
+        }
+    };
+
+    if res.status().is_success() {
+        println!("🥳 Successfully published your script.")
+    } else {
+        failure::bail!(
+            "⛔ Something went wrong! Status: {}, Details {}",
+            res.status(),
+            res.text()?
+        )
+    }
 
     Ok(())
 }
 
+fn build_js() -> Result<String, failure::Error> {
+    let mut f = fs::File::open("worker.js")?;
+    let mut buffer = String::new();
+    f.read_to_string(&mut buffer)?;
+    Ok(buffer)
```

Comment:
This function body can be replaced entirely with

```rust
std::fs::read_to_string("worker.js")
```

though, that returns an `io::Error`, so you *may* need

```rust
Ok(std::fs::read_to_string("worker.js")?)
```

Reply:
yesss, as said in air i don't know why but the docs i end up getting on google search never show this very good optimized way